util.hpp

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include <cassert>
+#include <memory>
+#include <string>
+#include <vector>
+#include <chrono>
+#include <iostream>
+#include <algorithm>
+#include <thread>
+
+using timer = std::chrono::high_resolution_clock;
+using usecs = std::chrono::microseconds;
+
+inline double delta_t(const timer::time_point t0, const timer::time_point t1) {
+  return (t1 - t0).count();
+}
+
+struct benchmark_parameters {
+  size_t threads, blocks;
+  size_t repetitions, epochs;
+  size_t array_size, array_size_per_kernel;
+  size_t slots, kernels_per_slot;
+};
+
+template<typename T>
+inline T* offset_parameter(T* ptr, size_t step, size_t idx) {
```

Comment:
Can we rename this to `offset_and_forward`. It adds a bit more clarity.

Reply:
Sure, that's a better name.